src/components/root_layout.tsx

```diff
@@ -26,47 +26,33 @@ const RootLayout: FC<RootLayoutProps> = ({ children }) => {
 // templateName set to anything other than the empty string while showBackButton is true indicates that the 
 // back button should send the user to the JobsView. templateName set to the empty string while 
 // showBackButton is true indicates that the back button should send the user to the Home screen.
-const [templateName, setTemplateName] = useState("");
 const [showBackButtonToTemplate, setShowBackButtonToTemplate] = useState(false);
-const [showBackButtonToHome, setShowBackButtonToHome] = useState(false);
+const [backUrl, setBackUrl] = useState("/");
 
 useEffect(() => {
     const regexPatternToHome = /^(.*?)\/app\//;
     const regexPatternToTemplate = /^.*?\/app\/([^\/]+)\/([^\/]+)$/;
     const toTemplateMatchResult = location.pathname.match(regexPatternToTemplate);
     if (toTemplateMatchResult) {
         setShowBackButtonToTemplate(true);
-        setShowBackButtonToHome(false);
         const [, capturedTemplateName] = toTemplateMatchResult;
-        setTemplateName(capturedTemplateName);
-    }  else if (regexPatternToHome.test(location.pathname)){
-        setShowBackButtonToHome(true);
-        setShowBackButtonToTemplate(false);
-        setTemplateName("");
+        setBackUrl("\/app\/"+capturedTemplateName);
+    } else if(regexPatternToHome.test(location.pathname)){
+        setShowBackButtonToTemplate(true);
```

Comment:
Wouldn't it be clearer to just call this state variable `showBackButton`? The back button doesn't always go to the template.